### ISSUE TITLE
baselayers : affichage fonds de plans L93

### ISSUE THREAD

**Issue by lecault** [geobretagne/mviewer#588]
**Description du bug / Describe the bug**

Je suis face à un problème récurrent mais là je m'y casse la tête je suis preneur d'un avis externe.

**Reproduire le bug / To Reproduce**

Etapes à suivre pour reproduire votre problème du type : 

1. Aller sur la page https://geobretagne.fr/mviewer/?config=/pub/sdis56/apps/visualisateur_test.xml
2. Les fonds de plans sont mal affichés

**Comportement attendu / Exptected behavior**

On ne devrait voir uniquement la pastille du fond de plan actif

**Screenshots**

![image](https://user-images.githubusercontent.com/5269430/160152443-8024739a-5dfb-4aeb-9ff0-ac382886d967.png)

**Informations complémentaires / Additional context**

Lien XML : https://geobretagne.fr/pub/sdis56/apps/visualisateur_test.xml
L'application est en Lambert 93 et a été conçue à partir de studio (puis retouches).
Merci !

**Comment by lecault** [comment]
Piste : sur cette appli j'ai des soucis pour faire un screenshot sur certains fonds de plan (Ortho, carte IGN, plan IGN) que je n'ai pas ailleurs. Exemple ici où ça marche https://kartenn.region-bretagne.fr/kartoviz/?config=apps/region/tmp/ign.xml

Voici le message d'erreur : SecurityError: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.

Je pense que les problèmes sont liés merci !

**Comment by Gaetanbrl** [comment]
Ce que tu veux c'est ne pas voir toutes les pastilles c'est ca ?

**Comment by lecault** [comment]
Oui, en mode gallery, on ne voit que la pastille du fond de carte actif.

**Comment by Gaetanbrl** [comment]
J'ai déjà eu ce problèle.
Dans ton cas, dès que je veut utiliser un de tes fond de plan de ton XML, j'ai la page d'erreur Mviewer.

Ex : 

`<baselayer visible="true" id="osm_google" thumbgallery="img/basemap/osm_google.png" title="GéoBretagne" label="OpenStreetMap" type="WMS" url="https://osm.geobretagne.fr/gwc01/service/wms" layers="osm:google" format="image/png" attribution="GéoBretagne. Données : les contributeurs d'<a href="https://www.openstreetmap.org/" target="_blank">OpenStreetMap </a>, <a href="https://www.openstreetmap.org/copyright" target="_blank">ODbL </a>"/>`


En supprimant l'attribution ca fonctionne.

Je pense que tu as 2 soucis : 

- Des doubles quotes dans une string qui commence et se finie par des doubles quottes (ca casse la chaine ca marche pas)
- On doit utiliser `&lt;` à la place de <

**Comment by lecault** [comment]
Merci de ton investigation Gaëtan.

Ecoute j'ai toujours le soucis ici avec des attributions vides :
https://geobretagne.fr/mviewer/?config=/pub/sdis56/apps/visualisateur_test2.xml
https://geobretagne.fr/pub/sdis56/apps/visualisateur_test2.xml

J'avoue qu'il me prend la tête ce xml. Si tu as un lien ou il marche, je suis preneur :-).

**Comment by Gaetanbrl** [comment]
En mode gallery, les vignettes ont une classe CSS `no-active` (sauf la baselayer affichée sur la carte qui a une classe `active`).
Mais de ce que je vois, cette class est : 
1. mal initialisée
2. ou mal prise en compte et parfois la vignette est visible malgré la class `no-active`

Exemple pour le cas 1 :
![image](https://user-images.githubusercontent.com/16317988/160357805-38cf666c-557c-4bd6-b9f0-0423652eed5d.png)

**Comment by Gaetanbrl** [comment]
Aussi, le problème ne semble venir qu'avec les couches IGN.

**Comment by lecault** [comment]
Je viens de trouver d'où vient l'erreur.
C'est lorsqu'on a fromcapacity="true" sur une des couches. J'ai essayé en mettant à false, les vignettes s'affichent bien.
Mais pas la carte vu qu'on veut du Lambert 93.

Et cela pour les données IGN.

**Comment by Gaetanbrl** [comment]
@lecault on a un fix pour cette solution ou on doit investiguer ?

**Comment by lecault** [comment]
On doit investiguer.
Après, cela me semble pas prioritaire. Cela ne concerne que les cartes dans une projection type Lambert93 avec des fonds de cartes IGN.

**Comment by lecault** [comment]
Je n'ai plus le souci ici sur les fonds ici via usage du WMS : https://geobretagne.fr/mviewer/?config=/pub/sdis56/apps/visualisateur.xml

Je clos
